### PR TITLE
Fix broken DXVA deinterlacing

### PR DESCRIPTION
**Pull request by Voyager1**
For DXVA, the deinterlacer was not turned on because the method setting now is always "automatic" and that value wasn't recognized. 

Remaining work: Remove "software blend" - I'll do that in a later PR.

### CODE REVIEW

**Comment by Voyager1** [comment]
updated PR - removed the "wrong" unnecessary commit. This is now only DXVA. @afedchin can you quickly review -thx

**Comment by FernetMenta** [comment]
During my tests I noticed an additional issue with regard to deinterlaing on Windows. If you force DXVA rendering, you don't get the SW deinterlacing options like "Deinterlacing" when dxva2 decoder is not used.

**Comment by Voyager1** [comment]
> If you force DXVA rendering, you don't get the SW deinterlacing options like "Deinterlacing"
I agree - somehow those options have been removed, but there's no reason why you couldn't have ffmpeg yadif deinterlacing, followed by a PresentSingle output to DXVA...
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/WinRenderer.cpp#L960

**Comment by Voyager1** [comment]
the only rationale I see for NOT enabling software-based deinterlacing is that DXVA is "typically" used for HD video, and there you really only need to limit the options to the DXVA HD processor. From that point of view I would discourage adding the software options (although they will work fine on SD).

**Comment by FernetMenta** [comment]
when doing sw decoding the sw deinterlacing options should be available. that is the case on all platforms regardless what renderer is chosen. making it dependent on randerer, pixel shader or dxva, is wrong.

**Comment by Voyager1** [comment]
sure but the decision on the options available is within the renderer. If we can get access to which decoder is used, we could do as you say...

**Comment by Voyager1** [comment]
@FernetMenta - I suggest we work on these suggestions in another PR. At least this one here fixes the most urgent issue, namely that there is no working deinterlacing when using DXVA. Any objections to pull this in?

**Comment by FernetMenta** [comment]
ok, let's have @afedchin have a final look at this one

**Comment by afedchin** [comment]
we should get rid of VS_INTERLACEMETHOD_DXVA_BEST and VS_INTERLACEMETHOD_DXVA_BOB at all.

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp [hunk 1]
```diff
@@ -599,9 +599,7 @@ bool CProcessorHD::Render(CRect src, CRect dst, ID3D11Resource* target, ID3D11Vi
     deinterlace_mode = (flags & RENDER_FLAG_FIELD0 || flags & RENDER_FLAG_FIELD1) ? VS_DEINTERLACEMODE_FORCE : VS_DEINTERLACEMODE_OFF;
   EINTERLACEMETHOD interlace_method = CMediaSettings::GetInstance().GetCurrentVideoSettings().m_InterlaceMethod;
```
Comment (afedchin) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp]: drop this also. it doesn't make sence anymore.

**Comment by afedchin** [comment]
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/WinRenderer.cpp#L1047
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp#L964
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp#L964

**Comment by afedchin** [comment]
this will be good for me if we drop this also https://github.com/xbmc/xbmc/pull/8770#discussion-diff-48945469R600

**Comment by Voyager1** [comment]
@afedchin - thanks I will do this later tonight

**Comment by afedchin** [comment]
@Voyager1 thanks.

**Comment by Voyager1** [comment]
@afedchin - done. Removing these options somehow also fixed the forced deinterlacer on DXVA.
jenkins build this please

File: xbmc/cores/IPlayer.h [hunk 2]
```diff
@@ -159,9 +159,6 @@ enum EINTERLACEMETHOD
   VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL=14,
   VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL_HALF=15,
   VS_INTERLACEMETHOD_DEINTERLACE_HALF=16,
-  VS_INTERLACEMETHOD_DXVA_BOB = 17,
```
Comment (FernetMenta) [inline xbmc/cores/IPlayer.h]: not sure if this is supposed to have gaps. did you check CGUIDialogVideoSettings::InitializeSettings what happens when adding something undefined to a spinner

```
    CSettingInt *settingInterlaceMethod = AddSpinner(groupVideo, SETTING_VIDEO_INTERLACEMETHOD, 16038, 0, static_cast<int>(videoSettings.m_InterlaceMethod), entries);
```

File: xbmc/video/dialogs/GUIDialogVideoSettings.cpp [hunk 3]
```diff
@@ -259,8 +259,6 @@ void CGUIDialogVideoSettings::InitializeSettings()
   entries.push_back(std::make_pair(16318, VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL_HALF));
   entries.push_back(std::make_pair(16317, VS_INTERLACEMETHOD_VDPAU_TEMPORAL_HALF));
   entries.push_back(std::make_pair(16314, VS_INTERLACEMETHOD_VDPAU_INVERSE_TELECINE));
-  entries.push_back(std::make_pair(16320, VS_INTERLACEMETHOD_DXVA_BOB));
```
Comment (FernetMenta) [inline xbmc/video/dialogs/GUIDialogVideoSettings.cpp]: I think we should add something like "undefined" or "depreciated" here.

Comment (Voyager1) [inline xbmc/video/dialogs/GUIDialogVideoSettings.cpp]: why? this is for the GUI only anyway, so the next available option would be picked... Furthermore the option can't be used in by code anyway.

Comment (Voyager1) [inline xbmc/video/dialogs/GUIDialogVideoSettings.cpp]: did you mean just comment out the line, with add'l comment saying it's deprecated?

Comment (FernetMenta) [inline xbmc/video/dialogs/GUIDialogVideoSettings.cpp]: is that so? I am not sure what a spinner does when selecting a value not added. if they are robust enough, then fine.

File: xbmc/cores/IPlayer.h (hunk 2 above)
Comment (Voyager1) [inline xbmc/cores/IPlayer.h]: there are gaps already - 10 is missing. This is just an enum, you're not even obliged to give them  values, only if you want to make sure they persist :-) This doesn't cause issues.

Comment (FernetMenta) [inline xbmc/cores/IPlayer.h]: ok fine. don't you think VS_INTERLACEMETHOD_MAX is misleading then? this shouldn't be needed then, right?

File: xbmc/video/dialogs/GUIDialogVideoSettings.cpp (hunk 3 above)
Comment (Voyager1) [inline xbmc/video/dialogs/GUIDialogVideoSettings.cpp]: we had this all the time with this spinner: suppose you have selected Software rendering and choose "Deinterlace". Then save the setting. Go back to acceleration and say "dxva rendering". Now the options in the spinner are all gone, and the selection of "Deinterlace" (saved in the db) doesn't cause issues. Instead, it goes back to "Automatic". I say they are robust enough.

File: xbmc/cores/IPlayer.h (hunk 2 above)
Comment (Voyager1) [inline xbmc/cores/IPlayer.h]: MAX seems to be used by the XML loader as a safeguard against reading incorrect values. But max itself isn't used as a valid option, so I think this is redundant...

Comment (FernetMenta) [inline xbmc/cores/IPlayer.h]: ok, go ahead, I was just wondering. imo a safeguard does not make much sense if gaps are allowed. but this a different story and not related to this pr.

**Comment by Voyager1** [comment]
build failure unrelated.

**Comment by afedchin** [comment]
@Voyager1 please make a backport for jarvis with only first commit

**Comment by Voyager1** [comment]
Will do sometime tomorrow @afedchin

**Pull request merged**